src/sfizz/Region.cpp

```diff
@@ -1026,24 +1026,45 @@ bool sfz::Region::registerNoteOff(int noteNumber, float velocity, float randValu
             keySwitched = true;
     }
 
-    const bool keyOk = keyRange.containsWithEnd(noteNumber);
-
     if (!isSwitchedOn())
         return false;
 
     if (!triggerOnNote)
         return false;
 
+    // Prerequisites
+
+    const bool keyOk = keyRange.containsWithEnd(noteNumber);
     const bool velOk = velocityRange.containsWithEnd(velocity);
     const bool randOk = randRange.contains(randValue);
-    bool releaseTrigger = (trigger == SfzTrigger::release_key);
+
+    if (!(velOk && keyOk && randOk))
+        return false;
+
+    // Release logic
+
+    if (trigger == SfzTrigger::release_key)
+        return true;
+
     if (trigger == SfzTrigger::release) {
         if (midiState.getCCValue(sustainCC) < config::halfCCThreshold)
-            releaseTrigger = true;
+            return true;
+
+        // If we reach this part, we're storing the notes to delay their release on CC up
+        // This is handled by the Synth object
+
+        const auto sameNoteTest = [noteNumber](const std::pair<int, float>& noteAndValue) {
+            return noteAndValue.first == noteNumber;
+        };
+
+        auto it = absl::c_find_if(delayedReleases, sameNoteTest);
+        if (it == delayedReleases.end())
+            delayedReleases.emplace_back(noteNumber, midiState.getNoteVelocity(noteNumber));
```

Comment:
do we force voice count to be <= 1 per note number ?

Reply:
You mean you want possibly multiple release samples per note? Maybe it could make sense for some instruments... Probably we should match what happens in sforzando.

Reply:
Can a single midi note have more that 1 Voice on it?
because that can be a desirable property.

Reply:
Yes sure, if multiple regions fire on the note trigger no problem. But this change is only for having multiple voices fire on a single region, which I think only happens conceptually if there is sustain and the region is a "release" region. Or maybe I misunderstand your question :slightly_smiling_face: